### DISCUSSION TITLE
TCP API hooks and pass-through connections

### DISCUSSION THREAD

**Discussion by EndUser509** [mitmproxy/mitmproxy#5455] [Ideas]
I was trying to write a small script that would print me the number of bytes sent and received over pass-through connections (` data.ignore_connection` set to `true` in `tls_clienthello()`). I naively assumed that the `tcp_*` hooks would be called for such a connection but it seems that is not the case.
Would it make sens to have these called? This would eventually require a set of UDP hooks similar to the TCP hooks for QUIC and the like so it may be a bigger thing.

**Comment by meitinger** [comment]
Actually, the QUIC draft PR #5435 already supports `tls_clienthello` and currently calls the `tcp_*` hooks for pass-through connections.
There is a discussion in PR #5414 how this should look like in the future, however.

If I understand your approach correctly, you would actually benefit from having unified `raw_*` hooks, as mentioned in https://github.com/mitmproxy/mitmproxy/pull/5414#issuecomment-1170315871. If so, maybe leave a comment there, perhaps this will keep the discussion going in that direction :)
